app/models/document.rb

```diff
@@ -206,10 +206,12 @@ def save!
       presented_document = DocumentPresenter.new(self)
       presented_links = DocumentLinksPresenter.new(self)
 
-      handle_remote_error do
+      success = handle_remote_error do
         Services.publishing_api.put_content(self.content_id, presented_document.to_json)
         Services.publishing_api.patch_links(self.content_id, presented_links.to_json)
       end
+
+      raise RecordNotSaved unless success
     else
       raise RecordNotSaved
     end
```

Comment:
Could we make the `#valid?` check a guard clause and then flatten out the method? i.e.

```
def save!
  raise RecordNotSaved unless self.valid?

  self.public_updated_at = Time.zone.now if self.update_type == 'major'
  # ...
end
```

I think it might be a bit cleaner. What do you think?